docs/source/delta-spark-connect.md

```diff
@@ -0,0 +1,124 @@
+  --conf "spark.sql.extensions=io.delta.sql.DeltaSparkSessionExtension" \
+  --conf "spark.sql.catalog.spark_catalog=org.apache.spark.sql.delta.catalog.DeltaCatalog" \
+  --conf "spark.connect.extensions.relation.classes=org.apache.spark.sql.connect.delta.DeltaRelationPlugin" \
+  --conf "spark.connect.extensions.command.classes=org.apache.spark.sql.connect.delta.DeltaCommandPlugin"
+```
+
+## How to use the Python Spark Connect Client with Delta
+
+The <Delta> Connect Python client is included in the same PyPi package as <Delta> Spark.
+
+1. `pip install pyspark==4.0.0`
+
+2. `pip install delta-spark==4.0.0`
+
+3. The usage is the same as Spark Connect. We just need to pass in a remote `SparkSession` (instead of a local one) to the `DeltaTable` API.
+
+An example:
+
+```python
+import io.delta.tables._
+import org.apache.spark.sql.functions._
+
+val deltaTable = DeltaTable.forName(spark, "my_table")
+deltaTable.toDF().show()
+
+// Update will not be supported in the first of Delta 4.0.
+deltaTable.update(
+  condition = expr("id % 2 == 0"),
+  set = Map("id" -> expr("id + 100")))
+```
+
+## How to use the Scala Spark Connect Client with Delta
+
+.. note:: These steps are going to be simplified with final Spark 4.0 and Delta 4.0
+
+Make sure you are using Java 17!
+
+1. `export SPARK_REMOTE="sc://localhost:15002"`
+
+2. `export _JAVA_OPTIONS="--add-opens=java.base/ALL-UNNAMED"`
+
+3. Download and install the [Coursier CLI](https://get-coursier.io/docs/cli-installation)
+
+4. Start the Spark Connect client version 4.0.0 First Preview:
+
+```bash
+cs launch --jvm zulu:17.0.11 --scala 2.13.13 -r \
+m2Local com.lihaoyi:::ammonite:3.0.0-M2 org.apache.spark::spark-connect-client-jvm:4.0.0-preview1 io.delta:delta-connect-client_2.13:4.0.0-preview1 io.delta:delta-connect-common_2.13:4.0.0-preview1 com.google.protobuf:protobuf-java:3.25.1 \
+--java-opt --add-opens=java.base/java.nio=ALL-UNNAMED \
+-M org.apache.spark.sql.application.ConnectRepl
+```
+
+An example:
+    
+```scala
+import io.delta.tables._
+import org.apache.spark.sql.functions._
+
+val deltaTable = DeltaTable.forName(spark, "my_table")
+deltaTable.toDF.show()
+
+// Update will not be supported in the first of Delta 4.0.
+deltaTable.update(
+condition = expr("id % 2 == 0"),
+set = Map("id" -> expr("id + 100")))
+```
+
+In the future, when [spark-connect-repl](https://spark.apache.org/docs/4.0.0-preview1/spark-connect-overview.html#use-spark-connect-for-interactive-analysis) moves to Spark 4.0 and above, we will use `spark-connect-repl` instead of `cs`.
+
+## Preview Limitations
+
+The feature introduces a limited set of supported operations in <Delta> 4.0 preview and expands it in <Delta> 4.0 and above.
+
+.. csv-table::
+:header: "Supported operations - Delta 4.0 preview", "Supported operations - Delta 4.0 release"
+
+"`forPath`","`forPath`"
+"`forName`","`forName`"
+"`toDF`","`toDF`"
+,"`alias`"
```

Comment:
By the way, didn't we also add support for that? I remember seeing this in both the scala and python clients.

Reply:
@andreaschat-db Once I finish testing this to be sure I will update it later, this is fine as discussed with @tdas